### PR TITLE
linagora/lgs/linshare/products/linshare-mobile-apps#590 LA-590: Export file in iOS presentation

### PR DESCRIPTION
**Pull request by manhduc-nguyen**
@hoangdat @imGok please review this PR

ezgif-4-57e48a81bcf6

### CODE REVIEW

File: lib/l10n/intl_fr.arb [hunk 1]
```diff
@@ -106,5 +106,27 @@
...
+  "@cancel": {
+    "type": "text",
+    "placeholders": {}
+  },
+  "downloading_file": "Downloading {fileName}",
+  "@downloading_file": {
+    "type": "text",
+    "placeholders": {
+      "fileName": {}
+    }
```
Comment (hoangdat) [inline lib/l10n/intl_fr.arb:130]: please add `fr` too.

File: lib/l10n/intl_ru.arb [hunk 2]
```diff
@@ -106,5 +106,27 @@
...
+  "@cancel": {
+    "type": "text",
+    "placeholders": {}
+  },
+  "downloading_file": "Downloading {fileName}",
+  "@downloading_file": {
+    "type": "text",
+    "placeholders": {
+      "fileName": {}
+    }
```
Comment (hoangdat) [inline lib/l10n/intl_ru.arb:130]: Create a ticket for each sprint to make all the text missing translation, we will request translate in pool. Thanks

File: lib/presentation/view/downloading_file/downloading_file_builder.dart [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  Key _key;
+  String _title = '';
+  String _content = '';
+  String _actionText = '';
+
+  DownloadingFileBuilder(this._context, this.cancelToken, this._appNavigation);
+
+  DownloadingFileBuilder key(Key key) {
+    _key = key;
```
Comment (hoangdat) [inline lib/presentation/view/downloading_file/downloading_file_builder.dart:53]: this `key` is not used anywhere

File: lib/presentation/widget/myspace/my_space_viewmodel.dart [hunk 4]
```diff
@@ -105,6 +129,27 @@ class MySpaceViewModel extends BaseViewModel {
     };
   }
 
+  ThunkAction<AppState> _exportFileAction(Document document, CancelToken cancelToken) {
+    return (Store<AppState> store) async {
+      await _downloadFileIOSInteractor.execute(document, cancelToken).then((result) => result.fold(
+              (failure) {
+                if (failure is DownloadFileIOSFailure &&
+                    !(failure.downloadFileException is CancelDownloadFileException)) {
```
Comment (hoangdat) [inline lib/presentation/widget/myspace/my_space_viewmodel.dart:137]: Separate to a new function

File: lib/presentation/widget/myspace/my_space_widget.dart [hunk 5]
```diff
@@ -361,23 +362,34 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
     return SizedBox.shrink();
   }
 
-  List<Widget> contextMenuActionTiles(Document document) {
+  List<Widget> contextMenuActionTiles(BuildContext context, Document document) {
+    if (Platform.isIOS) {
+      return [exportFileAction(context, document)];
+    }
     return [];
```
Comment (hoangdat) [inline lib/presentation/widget/myspace/my_space_widget.dart:369]: how about 
```
return [
  if (Platform.isIOS) exportFileAction(context, document);
];
```

File: pubspec.yaml [hunk 6]
```diff
@@ -84,6 +84,9 @@ dependencies:
   #filesize
   filesize: ^1.0.4
 
+  # share
+  share: ^0.6.5+4
```
Comment (hoangdat) [inline pubspec.yaml:88]: I think from now, we should not use `^`, it is risk for us too much.

Comment (manhduc-nguyen) [reply pubspec.yaml:88]: agree

**Comment by manhduc-nguyen** [comment]
@hoangdat pls check mixup commit

File: lib/l10n/intl_ru.arb (hunk 2 above)
Comment (manhduc-nguyen) [reply lib/l10n/intl_ru.arb:130]: done: [https://ci.linagora.com/linagora/lgs/linshare/products/linshare-mobile-apps/issues/605](https://ci.linagora.com/linagora/lgs/linshare/products/linshare-mobile-apps/issues/605)

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**